### PR TITLE
Issue 5787: Updated Basis Universal to v1.15 with rdo optimizations

### LINKED ISSUE #5787: Update Basis UASTC
Update basis library with RDO optimization support

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

**Comment by JCash** [comment]
Some numbers when compressing the [kodim/kodim03.png](http://r0k.us/graphics/kodak/kodim03.html). (The image was made into a power of two.
Percentace is of the the `kodim03.png` file size (522369 bytes)

            // # Best first, fast last
            // test_uastc(image, uastc_level=2, uastc_rdo_l=0.25, uastc_rdo_d=16384)   # 64.88%, 47.977 dB, 3.139894 s
            // test_uastc(image, uastc_level=1, uastc_rdo_l=1, uastc_rdo_d=16384)      # 52.31%, 45.047 dB, 2.310900 s
            // test_uastc(image, uastc_level=1, uastc_rdo_l=2, uastc_rdo_d=8192)       # 49.82%, 42.927 dB, 1.243244 s
            // test_uastc(image, uastc_level=0, uastc_rdo_l=3, uastc_rdo_d=8192)       # 43.79%, 40.481 dB, 0.778897 s

File: engine/texc/src/texc_enc_basis.cpp [hunk 1]
```diff
@@ -26,27 +26,74 @@ namespace dmTexc
...
+                case CL_NORMAL: comp_params.m_quality_level = 160; break;
+                case CL_HIGH:   comp_params.m_quality_level = 196; break;
+                case CL_BEST:   comp_params.m_quality_level = 256; break;
+                default:        comp_params.m_quality_level = 160; break;
             }
         }
         else {
 
-            switch(compression_level)
+            // # Best first, fast last
```
Comment (JCash) [inline engine/texc/src/texc_enc_basis.cpp:58]: The actual texc settings

File: engine/texc/src/texc_enc_basis.cpp [hunk 2]
```diff
@@ -72,6 +119,17 @@ namespace dmTexc
...
+        uint32_t w = texture->m_BasisImage.get_width();
+        uint32_t h = texture->m_BasisImage.get_height();
+        basisu::color_rgba* pixels = texture->m_BasisImage.get_ptr();
+
+        if (pixel_format == PF_R4G4B4A4) {
+            DitherRGBA4444((uint8_t*)pixels, w, h);
+        }
+        else if(pixel_format == PF_R5G6B5) {
+            DitherRGBx565((uint8_t*)pixels, w, h);
+        }
```
Comment (JCash) [inline engine/texc/src/texc_enc_basis.cpp:131]: I had forgot to dither these images

**Review by Jhonnyg — APPROVED** [review]
Looks good!

**Pull request merged**